### ISSUE TITLE
Glow stone canyon not generating in Terraforged

### ISSUE THREAD

**Issue by HalfBakedA** [nekomaster1000/Infernal-Expansion#227]
**Describe the Bug:**
Terraforged is a popular mod that redoes Minecraft's generation within the overworld, This mod doesn't effect the nether at all, and all generation added by the infernal expansion works within this mod, but for some reason the new glow stone canyon biome does not generate at all. I'm not sure why this is happening as other custom biomes from other mods seem to work, but not this one?

**How to Reproduce:**
1. Download Terraforged and put it in your mods folder (https://www.curseforge.com/minecraft/mc-mods/terraforged)
2. When creating a new world, click new options and change the world type until it says Terraforged.
3. Create world, and go into the nether (/execute in minecraft:the_nether run tp @p 0 90 0)
4. Try to locate a Glowstone canyon. (/locatebiome infernalexp:glowstone_canyon)

**Version Information:**
 - Operating System: Windows 10
 - Mod Version: 2.0.0
 - Minecraft Version: 1.16.5
 - Forge Version: Latest
- Terraforged Version: 0.2.13

**Additional Context:**
I have found that the biome does still spawn properly when using any other world type with terraforged installed, its just any variant of terraforged's world type.

**Comment by evanbones** [comment]
Can confirm, weirdly enough can be worked around when changing the worldtype for the nether to quark:realistic

**Comment by Dariensg** [comment]
Hi! I just wanted to give you both a quick update on this since you seem to have found the same thing we discovered after talking with other mod developers. Terraforged forcefully grabs the Vanilla BiomeProvider for the Nether and End on their World Type. To put it simply, this means that no Modded biomes can spawn in either of those dimensions. They do have implemented the ability to select other mods' biome providers for the other dimensions upon world creation. The only problem is that those Biome Providers will only show up if that mod has a custom world type registered. As of right now, we don't have a world type. We are aware that Oh the Biomes You'll Go have created their own custom world type and then hidden it from the world creation menu specifically for the sake of use with Terraforged. We are considering our possible solutions and will update you when we've decided how to handle this issue. For now, we are aware that the BYG Nether biome world type and, apparently, the quark Realistic one will bypass this issue until we decide upon a fix. Until then, we are sorry for the inconvenience.

**Comment by HalfBakedA** [comment]
Thanks for the information! I have recently been making a modpack for friends and I was really upset Terraforged seemed to break it. Going to look into Quark for potential fixes. I do want to note however that I've seen custom biomes work in the end with terraforged, as at the same time I've had Team Abnormal's Endergetic expansion on, and their custom biome "the poise forest" seems to work uninterrupted by it

**Comment by Dariensg** [comment]
That's interesting to hear! I could be wrong about how the End works as far as BiomeProviders go. I haven't worked with that code directly. Part of the problem is that Forge is a little bit of a mess when it comes to Nether biomes, which causes issues with all of these mods. We intentionally make an effort to grab all other mods' biomes in our world generation process, which needs to be done for compat between two Nether mods. It's possible the End doesn't have this same issue, but I'll look into their code to see if they seem to have found a way around it after all! Thank you for the pointer!

**Comment by Dariensg** [comment]
Hey! Just wanted to let you know that we decided to go ahead and make a custom World Type (with nothing special) for use with TerraForged. It's PRed right now. Once that gets merged, in our next Release version (which should be very soon) we will be compatible with TerraForged! All  you'll have to do is select "Customize" under the TerraForged World Type while creating the world, go to the second page, and under "Nether," select `infernalexp:compat_world_type`. I've tested and this works and includes BYG (and presumably BOP although I haven't tested this) biomes as well in the Nether! Thanks for bringing this to our attention!
